### ISSUE TITLE
Option to display images only when accepted

### ISSUE THREAD

**Issue by PaulFreund** [vector-im/riot-web#4286]
It would be a good addition to add an user option to display uploaded images only after the user has "accepted" them, for example by clicking a download button and dont show a preview. This has the benefit of deciding which images you want to see and it can be easier on the data for mobile users

Might be related to #2466

**Comment by ara4n** [comment]
Yeah, good idea. One might also want to tag the event as NSFW or similar to stop it from displaying thumbnails by default.

**Comment by rodneyrod** [comment]
Yeah this is something I want to bring up.
Also even when you disable in line previews pictures still show up which isn't the best in a SFW situation.

**Comment by PureTryOut** [comment]
Please, please give us an option to disable the thumbnail by default. In some chats people like to spam images every once and while, and besides some being NSFW, it just makes reading back history really annoying due to having to scroll too much before you can read anything agian.

**Comment by plusgut** [comment]
@PureTryOut what would you like to see instead? Just the link?

**Comment by rodneyrod** [comment]
@plusgut could do what telegram does, a blurred low res thumbnail of the image, or a blank thumbnail with 'click to download filename' or something like that.

**Comment by PureTryOut** [comment]
I'd like a link myself, that when clicked on opens it like what happens now if you click on the image. This way scrollback is no problem.
Maybe the other methods @rodneyrod mentioned could be added as an option.

**Comment by rodneyrod** [comment]
Maybe a mechanism to cache the image but blur it again when it's offscreen, that might be better suited in a dedicated SFW mode or something.
Or an option to hide the image again after it's previewed.

**Comment by PureTryOut** [comment]
It's not just about SFW and NSFW though; having an image inline files up a big part of your screen. Sure making the image smaller would already help, but it could be eliminated entirely by just providing a link. But the more options the better!

**Comment by BurnyBoi** [comment]
Per my comment on #8790, I think this should apply to videos as well as images. Riot displays the first frame of the video behind the play button, so this is also something I would like to be able to hide by default when I'm at work.
